cmd/prometheus/main.go

```diff
@@ -307,6 +308,10 @@ func (c *flagConfig) setFeatureListOptions(logger *slog.Logger) error {
 		return errors.New("cannot enable otlp-deltatocumulative and otlp-native-delta-ingestion features at the same time")
 	}
 
+	if c.web.NativeOTLPDeltaIngestion && !c.web.EnableTypeAndUnitLabels {
+		return errors.New("cannot enable otlp-native-delta-ingestion feature without enabling type and unit labels feature")
+	}
+
```

Comment:
Hmmm, not sure where in the proposal says that these two should be used together 🤔 

I'm not saying we shouldn't be doing that, I just don't remember where we came to that conclusion 😛

Reply:
After reading the proposal a bit more, it actually says the opposite:

```
This is similar to the approach taken for type and unit labels in PROM-39. However, since these are new labels being added, there is not a strong dependency so does not require `--enable-feature=type-and-unit-labels` to be enabled.
```